drivers/wifi/infineon/cyw43xxx_drv.c

```diff
@@ -0,0 +1,396 @@
+		case WLC_E_DISASSOC_IND:
+			(void)net_if_down(infineon_cyw43xxx_wifi_iface);
+			break;
+
+		default:
+			break;
+		}
+	}
+}
+
+static void eth_infineon_cyw43xxx_init(struct net_if *iface)
+{
+	const struct device *dev = net_if_get_device(iface);
+	struct infineon_cyw43xxx_wifi_runtime *dev_data = dev->data;
+
+	dev_data->iface = iface;
+	infineon_cyw43xxx_wifi_iface = iface;
+
+	/* Start interface when we are actually connected with WiFi network */
+	(void)net_if_flag_set(iface, NET_IF_NO_AUTO_START);
+
+	/* Read WLAN MAC Address */
+	(void)whd_wifi_get_mac_address(cyw43xxx_if, &cyw43xxx_if->mac_addr);
+	(void)memcpy(&dev_data->mac_addr, &cyw43xxx_if->mac_addr, sizeof(cyw43xxx_if->mac_addr));
+
+	/* Assign link local address. */
+	(void)net_if_set_link_addr(iface,
+				   dev_data->mac_addr, 6, NET_LINK_ETHERNET);
+
+	(void)ethernet_init(iface);
+}
+
+#if defined(CONFIG_NET_STATISTICS_ETHERNET)
+static struct net_stats_eth *eth_infineon_cyw43xxx_stats(const struct device *dev)
+{
+	return &(((struct infineon_cyw43xxx_wifi_runtime *)dev->data)->stats);
+}
+#endif
+
+static int eth_infineon_cyw43xxx_dev_init(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+	cy_rslt_t ret = CY_RSLT_SUCCESS;
+	cyhal_gpio_t sdio_cmd =
+		DT_GET_CYHAL_GPIO_FROM_DT_GPIOS(DT_WIFI_SDIO_NODE, wifi_sdio_cmd_gpios);
+
+	cyhal_gpio_t sdio_clk =
+		DT_GET_CYHAL_GPIO_FROM_DT_GPIOS(DT_WIFI_SDIO_NODE, wifi_sdio_clk_gpios);
+
+	cyhal_gpio_t sdio_d0 =
+		DT_GET_CYHAL_GPIO_FROM_DT_GPIOS(DT_WIFI_SDIO_NODE, wifi_sdio_d0_gpios);
+
+	cyhal_gpio_t sdio_d1 =
+		DT_GET_CYHAL_GPIO_FROM_DT_GPIOS(DT_WIFI_SDIO_NODE, wifi_sdio_d1_gpios);
+
+	cyhal_gpio_t sdio_d2 =
+		DT_GET_CYHAL_GPIO_FROM_DT_GPIOS(DT_WIFI_SDIO_NODE, wifi_sdio_d2_gpios);
+
+	cyhal_gpio_t sdio_d3 =
+		DT_GET_CYHAL_GPIO_FROM_DT_GPIOS(DT_WIFI_SDIO_NODE, wifi_sdio_d3_gpios);
+
+	ret = cyhal_sdio_init(&sdio_obj, sdio_cmd, sdio_clk,
+			      sdio_d0, sdio_d1, sdio_d2, sdio_d3);
+	if (ret != CY_RSLT_SUCCESS) {
+		LOG_ERR("cyhal_sdio_init failed ret = %d \r\n", ret);
+	}
+
+	k_tid_t tid = k_thread_create(&cyw43xxx_wifi_event_thread,
+				      cyw43xxx_wifi_event_stack,
+				      CONFIG_CYW43XXX_WIFI_EVENT_TASK_STACK_SIZE,
+				      (k_thread_entry_t)infineon_cyw43xxx_event_task,
+				      NULL, NULL, NULL,
+				      CONFIG_CYW43XXX_WIFI_EVENT_TASK_PRIO,
+				      K_INHERIT_PERMS, K_NO_WAIT);
+
+	k_thread_name_set(tid, "cyw43xxx_event");
+
+	ret = cybsp_wifi_init_primary_extended(&cyw43xxx_if, NULL, NULL, NULL, NULL);
+	if (ret != CY_RSLT_SUCCESS) {
+		LOG_ERR("cybsp_wifi_init_primary_extended failed ret = %d \r\n", ret);
+	}
+	ret = whd_management_set_event_handler(cyw43xxx_if, sta_link_events,
+					       link_events_handler, NULL,
+					       &sta_event_handler_index);
+	if (ret != CY_RSLT_SUCCESS) {
+		LOG_ERR("whd_management_set_event_handler failed ret = %d \r\n", ret);
+	}
+
+	return ret;
+}
+
+static struct infineon_cyw43xxx_wifi_runtime eth_data;
+static const struct ethernet_api eth_infineon_cy43xxx_apis = {
```

Comment:
IMO it'd be better to use [struct net_wifi_mgmt_offload](https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/net/wifi_mgmt.h#L301) - it's an extension over `struct ethernet_api` and allows to implement Wi-Fi mgmt functionalities, see how [esp32 does it](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/wifi/esp32/src/esp_wifi_drv.c#L618).

That way you'll be able to use standard Wi-Fi mgmt/shell with your driver and would not need a custom sample.

Reply:
Hi @rlubos,

I am re-working our driver to use net_wifi_mgmt_offload feature. I see that esp32 is using one net_if *esp32_wifi_iface for STA and AP. (https://github.com/zephyrproject-rtos/zephyr/blob/33b116407b03883efd889f75d69f5bc753b7ade3/drivers/wifi/esp32/src/esp_wifi_drv.c#L31)
Does this means that STA/AP can't work simultaneous?

Reply:
added net_wifi_mgmt_offload.